test/metabase/api/dashboard_test.clj

```diff
@@ -2047,36 +2047,82 @@
+      (testing "it only returns search matches"
+        (let-url [url (chain-filter-search-url dashboard-id "abc" "red")]
+          (is (= {:values          ["Red Medicine"]
+                  :has_more_values false}
+                 (mt/user-http-request :rasta :get 200 url))))))
+
+    (testing "field selection should compatible with field-id from /api/table/:card__id/query_metadata"
+      ;; FE use the id returned by /api/table/:card__id/query_metadata
+      ;; for the `values_source_config.value_field`, so we need to test to make sure
+      ;; the id is a valid field that we could use to retrieve values.
+      (mt/with-temp*
+        [;; card with agggregation and binning columns
+         Card [{mbql-card-id :id}
+               (merge (mt/card-with-source-metadata-for-query
+                        (mt/mbql-query venues {:limit 5
+                                               :aggregation [:count]
+                                               :breakout [[:field %latitude {:binning {:strategy :num-bins :num-bins 10}}]]}))
+                      {:name        "MBQL question"
+                       :database_id (mt/id)
+                       :table_id    (mt/id :venues)})]
+         Card [{native-card-id :id}
+               (merge (mt/card-with-source-metadata-for-query
+                        (mt/native-query {:query "select name from venues;"}))
+                      {:name        "Native question"
+                       :database_id (mt/id)
+                       :table_id    (mt/id :venues)})]]
+
+        (let [mbql-card-fields   (card-fields-from-table-metadata mbql-card-id)
+              native-card-fields (card-fields-from-table-metadata native-card-id)
+              fields->parameter  (fn [fields card-id]
+                                   (for [{:keys [id field_ref name]} fields]
+                                     {:id                   (format "id_%s" name)
+                                      :type                 "category"
+                                      :name                 name
+                                      :values_source_type   "card"
+                                      :values_source_config {:card_id     card-id
+                                                             :value_field (if (number? id)
+                                                                            field_ref
+                                                                            id)}}))
+              parameters         (concat
+                                   (fields->parameter mbql-card-fields mbql-card-id)
+                                   (fields->parameter native-card-fields native-card-id))]
+          (mt/with-temp Dashboard [{dash-id :id} {:parameters parameters}]
+            (doseq [param parameters]
+              (let-url [url (chain-filter-values-url dash-id (:id param))]
+                (is (some? (mt/user-http-request :rasta :get 200 url)))))))))))
```

Comment:
`some?` feels like a pretty weak predicate, but I guess if there was an issue it'd be nil/empty or we'd get a server error.

Is there value in doing a `schema=` with some/all of these fields from `card->virtual-table`?

```
    (cond-> {:id               (str "card__" (u/the-id card))
             :db_id            (:database_id card)
             :display_name     (:name card)
             :schema           (get-in card [:collection :name] (root-collection-schema-name))
             :moderated_status (:moderated_status card)
             :description      (:description card)}
      include-fields? (assoc :fields (card-result-metadata->virtual-fields (u/the-id card)
```

Reply:
there tests are meant to test compatibility only, so I think it's just redundant to test the output.

we had output tests for this in `metabase.models.params.card-values-test`